### DISCUSSION TITLE
Parcels syntax - including particle when field sampling

### DISCUSSION THREAD

**Discussion by jwongala** [Parcels-code/Parcels#1576] [Q&A]
Hi, I have a question about parcel syntax (unsure if this is the right word) and using status codes. Please let me know if I should separate these two questions. 

I am trying to get the temperature at a particle's location using this code: `temp1 = fieldset.T[particle]`. What does `particle` mean in this context? Is it a unique ID for an individual particle, or does it get the temperature for all particles in the upset? 

Also, after doing calculations during each timestep, I would like to find the temperature and salinity at the particle's new lat and lon and assign it to the `pset`. I am using this code, but it's not working, and I am unsure why `particle.temp = fieldset.T [time + particle.dt, dep1, particle.lat, particle.lon]`. I think this is related to not understanding what `particle` means, but I am unsure. Below is the end of the code where I assign the new lat, lon, temp, and salt at the particles new location

`particle.lon += (u1 + 2*u2 + 2*u3 + u4) / 6. * particle.dt`
`particle.lat += (v1 + 2*v2 + 2*v3 + v4) / 6. * particle.dt`
`particle.salt = fieldset.S[time + particle.dt, dep1, particle.lat, particle.lon]`
`particle.temp = fieldset.T[time + particle.dt, dep1,  particle.lat, particle.lon]`

**Comment by erikvansebille** [comment]
Dear @jwongala, first of all: please indeed separate different questions/issues into different posts; because that is the only way we can keep this resource manageable. You can of course refer to another post in your question/comments.

> I am trying to get the temperature at a particle's location using this code: temp1 = fieldset.T[particle]. What does particle mean in this context? Is it a unique ID for an individual particle, or does it get the temperature for all particles in the upset?

The difference between `value = fieldset.U[time, particle.depth, particle.lat, particle.lon]` and `value = fieldset.U[particle]` is explained in https://docs.oceanparcels.org/en/latest/examples/tutorial_jit_vs_scipy.html#Further-digging-into-Scipy-mode:-adding-particle-keyword-to-Field-sampling. In short, they are completely equivalent but the second version is slightly faster in Scipy-mode.

**Reply by jwongala** [reply]
Hello, @erikvansebille. My apologies, and thank you for letting me know! I will create separate discussion posts for my other two questions. 

Thanks! This makes sense with your explanation and the link you shared. I am using Scipy particles so I think I understand why my second related question isn't working. Mahalo!
